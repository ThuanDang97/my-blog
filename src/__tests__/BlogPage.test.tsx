// Components
import PageBlogList, {
  getStaticPaths,
  getStaticProps,
} from '@pages/blog/[page]'

// Utils
import { fireEvent, render, screen } from '@utils/testUtils'

// Mocks Data
import { MOCK_DATA_PAGE } from '@constants/__mock__/mockData'

jest.mock('@services/restClient', () => ({
  getDataContent: jest.fn(() => {
    return MOCK_DATA_PAGE
  }),
}))

jest.mock('next/router', () => ({
  useRouter: () => ({
    push: jest.fn(),
    back: jest.fn(),
  }),
}))

describe('Page Blog', () => {
  test('Page Blog matches DOM Snapshot', () => {
    const { container } = render(<PageBlogList posts={MOCK_DATA_PAGE} />)

    expect(container).toMatchSnapshot()
  })

  test('getStaticPropsPaths should return props', async () => {
    const result = await getStaticPaths({})

    expect(result).toStrictEqual({
      paths: [{ params: { page: `${MOCK_DATA_PAGE.length - 1}` } }],
      fallback: false,
    })
  })

  test('getStaticProps should return props ', async () => {
    const result = await getStaticProps({})

    expect(result).toStrictEqual({
      props: {
        posts: MOCK_DATA_PAGE,
      },
      revalidate: 60,
    })
  })

  test('should router math path after click button', () => {
    render(<PageBlogList posts={MOCK_DATA_PAGE} />)

    const olderButton = screen.getByRole('button', {
      name: /Older posts/i,
    })

    const newerButton = screen.getByRole('button', {
      name: /Newer posts/i,
    })

    fireEvent.click(olderButton)
    fireEvent.click(newerButton)

    expect(olderButton).toBeTruthy()
    expect(newerButton).toBeTruthy()
  })
})
