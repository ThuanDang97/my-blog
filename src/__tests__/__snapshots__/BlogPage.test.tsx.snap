// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Blog Page Blog matches DOM Snapshot 1`] = `
<div>
  <header
    class="MuiBox-root css-1p2cbqg"
  >
    <div
      class="MuiBox-root css-kp6j5k"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-9tjvhq-MuiContainer-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1pidjnp-MuiTypography-root"
        >
          <a
            href="/"
          >
            BEAUTYONCODE
          </a>
        </h1>
        <div
          class="MuiBox-root css-7w1jqr"
        >
          <ul
            class="MuiList-root MuiList-padding css-1mq3yj5-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/"
              >
                HOME
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/category/most-views"
              >
                Most views
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/category/short-posts"
              >
                Short Posts
              </a>
            </li>
          </ul>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-15on3wz-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1unfims-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div>
          loading.....
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-8uxnzh"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <nav
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-7xpsn0-MuiTypography-root"
              >
                <a
                  href="/"
                >
                  HOME
                </a>
              </p>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-hiwasb-MuiTypography-root"
              >
                <a
                  href="/blog"
                >
                  blog
                </a>
              </p>
            </li>
          </ol>
        </nav>
        <h2
          class="MuiTypography-root MuiTypography-h2 css-zd04td-MuiTypography-root"
          data-testid="page-title"
        />
      </div>
    </div>
  </header>
  <main
    class="MuiBox-root css-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-c731gg-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-x5eu6"
      >
        <article
          class="MuiBox-root css-zelrb1"
        >
          <figure
            class="MuiBox-root css-1defjo5"
          >
            <img
              alt="food-2203732 1280"
              data-nimg="1"
              decoding="async"
              height="500"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F6llpDgAKQDU0CzC8qJWeFh%2Fc31582e70f63c105e9c64f06b55af3cc%2Ffood-2203732_1280.png&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F6llpDgAKQDU0CzC8qJWeFh%2Fc31582e70f63c105e9c64f06b55af3cc%2Ffood-2203732_1280.png&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F6llpDgAKQDU0CzC8qJWeFh%2Fc31582e70f63c105e9c64f06b55af3cc%2Ffood-2203732_1280.png&w=3840&q=75 2x"
              style="color: transparent; max-width: 100%; max-height: 60%;"
              width="1280"
            />
            <div
              class="MuiBox-root css-1dzd2um"
            >
              <h2
                class="MuiTypography-root MuiTypography-h2 css-1r4hp4x-MuiTypography-root"
              >
                <a
                  href="/cac-bai-viet-ngan-phan-1"
                >
                  Các bài viết ngắn – phần 1
                </a>
              </h2>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiBox-root css-h0kv6j"
                >
                  <p>
                    .gitkeep File &lt;strong&gt;.gitkeep&lt;/strong&gt; thường được biết đến như là cách để có thể commit một thư mục trống lên trong quá trình thiết kế cấu trúc thư mục cho dự án của bạn. Ví dụ cấu trúc một website đơn…


                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-r6opal-MuiTypography-root"
                >
                  380 views
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1ngiwr2-MuiTypography-root"
                >
                  <a
                    href="/cac-bai-viet-ngan-phan-1"
                  >
                    CONTINUE READING...
                  </a>
                </p>
              </div>
            </div>
          </figure>
        </article>
        <article
          class="MuiBox-root css-zelrb1"
        >
          <figure
            class="MuiBox-root css-1defjo5"
          >
            <img
              alt="food-2203732 1280"
              data-nimg="1"
              decoding="async"
              height="500"
              loading="lazy"
              src="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F6llpDgAKQDU0CzC8qJWeFh%2Fc31582e70f63c105e9c64f06b55af3cc%2Ffood-2203732_1280.png&w=3840&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F6llpDgAKQDU0CzC8qJWeFh%2Fc31582e70f63c105e9c64f06b55af3cc%2Ffood-2203732_1280.png&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.ctfassets.net%2Font2c8ajcvde%2F6llpDgAKQDU0CzC8qJWeFh%2Fc31582e70f63c105e9c64f06b55af3cc%2Ffood-2203732_1280.png&w=3840&q=75 2x"
              style="color: transparent; max-width: 100%; max-height: 60%;"
              width="1280"
            />
            <div
              class="MuiBox-root css-1dzd2um"
            >
              <h2
                class="MuiTypography-root MuiTypography-h2 css-1r4hp4x-MuiTypography-root"
              >
                <a
                  href="/cac-bai-viet-ngan-phan-2"
                >
                  Các bài viết ngắn – phần 2
                </a>
              </h2>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiBox-root css-h0kv6j"
                >
                  <p>
                    .gitkeep File &lt;strong&gt;.gitkeep&lt;/strong&gt; thường được biết đến như là cách để có thể commit một thư mục trống lên trong quá trình thiết kế cấu trúc thư mục cho dự án của bạn. Ví dụ cấu trúc một website đơn…


                  </p>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-r6opal-MuiTypography-root"
                >
                  380 views
                   
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1ngiwr2-MuiTypography-root"
                >
                  <a
                    href="/cac-bai-viet-ngan-phan-2"
                  >
                    CONTINUE READING...
                  </a>
                </p>
              </div>
            </div>
          </figure>
        </article>
        <div
          class="MuiBox-root css-1ufrzn0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1ck34rp-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            Older posts
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1vmgmr7-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Newer posts
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowForwardIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-1kj6f6k"
      >
        <aside
          class="MuiBox-root css-3ok2v2"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-w1deiu-MuiTypography-root"
          >
            Recent Post
          </h2>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ejib7g-MuiListItem-root"
            >
              <a
                href="/cac-bai-viet-ngan-phan-1"
              >
                Các bài viết ngắn – phần 1
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ejib7g-MuiListItem-root"
            >
              <a
                href="/cac-bai-viet-ngan-phan-2"
              >
                Các bài viết ngắn – phần 2
              </a>
            </li>
          </ul>
        </aside>
        <aside
          class="MuiBox-root css-3ok2v2"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-w1deiu-MuiTypography-root"
          >
            Category
          </h2>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ejib7g-MuiListItem-root"
            >
              <a
                href="/blog"
              >
                All
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ejib7g-MuiListItem-root"
            >
              <a
                href="/category/programming"
              >
                Programming
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ejib7g-MuiListItem-root"
            >
              <a
                href="/category/short-posts"
              >
                Short Posts
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-ejib7g-MuiListItem-root"
            >
              <a
                href="/category/most-views"
              >
                Most Views
              </a>
            </li>
          </ul>
        </aside>
      </div>
    </div>
  </main>
</div>
`;
